src/index.ts

```diff
@@ -1,61 +1,48 @@
-import { Context, Hono } from "hono";
+import { Hono } from "hono";
 import { Error, ServiceAuthType } from "./types";
 import { streamResponse } from "./utils";
-import { Bindings } from "./bindings";
+import { AppContext, Bindings, Variables } from "./bindings";
 import HeaderUtils from "./header_utils";
 import analytics from "./middleware/analytics";
+import validate from "./middleware/validate";
 import cron from "./crons/cron";
 
-export const app = new Hono<{ Bindings: Bindings }>();
+export const app = new Hono<{ Bindings: Bindings; Variables: Variables }>();
 
 app.use("*", analytics());
+app.use("*", validate());
```

Comment:
just so i understand... it knows to use validate cause we use app.use here. is the order always analytics and then it validates and then it does the app.all?

Reply:
Yes, that's correct. It executes in the order it is written.